code-challenges.test.js

```diff
@@ -10,52 +12,127 @@
+            theseContain.push(check[j])
+        }
+    }
+    return theseContain
+}
 
+// GREEN
 
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
+// describe a function called fullHouse that takes in an array of 5 numbers and determines whether or not the array is a “full house”
 
+describe("fullHouse", () => {
+    let hand1 = [5, 5, 5, 3, 3]
+    let expectedOutput1 = true
+    let hand2 = [5, 5, 3, 3, 4]
+    let expectedOutput2 = false
+    let hand3 = [5, 5, 5, 5, 4]
+    let expectedOutput3 = false
+    it("takes in an array of 5 numbers and determines whether or not the array is a full house", () => {
+        expect(fullHouse(hand1)).toEqual(expectedOutput1)
+        expect(fullHouse(hand2)).toEqual(expectedOutput2)
+        expect(fullHouse(hand3)).toEqual(expectedOutput3)
+    })
+})
 
+// RED
 
 // b) Create the function that makes the test pass.
+
+// create a function called fullHouse that takes in an array and determines if there is a full house
+const fullHouse = (hand) => {
+    // iterate through the array, and push values into three arrays based on whether they match or not
+    // Start with this having the first value in the input array loaded into as a default to give the others a comparison start point
+    let valueMatch1 = [hand[0]]
+    // if either of these have a length of two and three then it is a full house
+    let valueMatch2 = []
+    // if this array is empty then there was only two unique values passed to the function
+    let doesNotMatch = []
+    for (let i = 1; i < hand.length; i++) {
+        // if the value matches the value then put it into the array
+        if (hand[i] === valueMatch1[0]) {
+            valueMatch1 = [...valueMatch1, hand[i]]
+        } else if (valueMatch2.length === 0) {
+            valueMatch2 = [hand[i]]
+        } else if (hand[i] === valueMatch2[0]) {
+            valueMatch2 = [...valueMatch2, hand[i]]
+        } else {
+            doesNotMatch = [...doesNotMatch, hand[i]]
+        }
+    }
+    // Now that we have placed our values into their matching arrays we determine if there is a full house. This is checked by having an empty doesNotMatch array, ie there are two or less different values passed into the function, and if valueMatch1 has either 2 or 3 items in it, which combined with an empty doesNotMatch array means valueMatch2 must have either 3 or 2 items in it, respectively, which satisfies the conditions of a full house.
+    return (doesNotMatch.length === 0 && valueMatch1.length === (3 || 2)) ? 
+        true : false
+}
+
```

Comment:
`valueMatch1.length === (3 || 2)` very cool set up here. I like your approach to this problem. Very systematic.